test/integration/persistent_bucket.py

```diff
@@ -0,0 +1,92 @@
+######################################################################
+#
+# File: test/integration/persistent_bucket.py
+#
+# Copyright 2024 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+import hashlib
+import os
+from dataclasses import dataclass
+from functools import cached_property
+from test.integration.helpers import BUCKET_NAME_LENGTH, Api
+
+import backoff
+from b2sdk.v2 import Bucket
+from b2sdk.v2.exception import DuplicateBucketName, NonExistentBucket
+
+PERSISTENT_BUCKET_NAME_PREFIX = "constst"
+
+
+@dataclass
+class PersistentBucketAggregate:
+    bucket_name: str
+    subfolder: str
+
+    @cached_property
+    def virtual_bucket_name(self):
+        return f"{self.bucket_name}/{self.subfolder}"
+
+
+@backoff.on_exception(backoff.expo, Exception, max_tries=3, max_time=10)
+def delete_all_files(bucket: Bucket):
+    all_items = list(bucket.ls(recursive=True))
+    for item, _ in all_items:
+        bucket.delete_file_version(item.id_, item.file_name)
+
+
+@backoff.on_exception(backoff.expo, Exception, max_tries=3, max_time=10)
+def delete_files(bucket: Bucket, subfolder: str):
```

Comment:
```suggestion
def delete_files(bucket: Bucket, subfolder: str | None = None):
```
seems like these two methods should be combined

Reply:
Seems to me like these calls are going:
a) slow down tests (so +++ cost on development side & GHA)
b) incur extra cost on B2 API side - but I guess that doesn't matter much for us

While simply waiting for lifecycle rules to do their thing, also means some extra B2 storage costs, but is much faster since we have to do nothing.

Reply:
Even though the exec time and the number of buckets used dropped as compared to the state before, I was clearly overcorrecting by clearing the subfolders between cases, trying to fully mimic the behavior of recreated buckets. Clearly redundant.

Reply:
soooo this code is gonna be removed?

Reply:
Thought I did already, done now